### ISSUE TITLE
Macros do not persist despite gdef and globalGroup

### ISSUE THREAD

**Issue by Recmo** [KaTeX/KaTeX#2513]
> Before reporting a bug
- [x] Check [common issues](https://katex.org/docs/issues.html).
- [x] Check the bug is reproducible in [the demo](https://katex.org). If not, check KaTeX is up-to-date and installed correctly.
- [x] Search for [existing issues](https://github.com/KaTeX/KaTeX/issues).

**Describe the bug:**
Macros are forgotten between invocations of `renderToString` (and possibly `render`), even when `globalGroup` is set and `\gdef` is used.

```javascript
katex.renderToString(String.raw`\gdef\foo{1}`);
katex.renderToString(String.raw`\foo`); // Fails
```

```javascript
katex.renderToString(String.raw`\gdef\foo{1}`, { globalGroup: true });
katex.renderToString(String.raw`\foo`); // Fails
```

**To Reproduce:**
1. Go to 'https://katex.org.'
2. Open javascript console
3. Enter either of the two examples above

**Expected behavior:**
Foo macro to expand because it was defined globally.

**Screenshots:**

**Environment (please complete the following information):**
 - KaTeX Version: 0.12.0
 - Device: Desktop
 - OS: MacOS 10.15.6 (19G2021)
 - Browser: Firefox
 - Version: 80.0.1 (64-bit)

**Additional context:**
I'm trying to define some globally available macros that require arguments. Doing this through the `macros` option seems extremely complicated and I can not find a working example, I'd much prefer parsing some `\gdef` statements.

**Comment by Recmo** [comment]
After digging some more, the following seems to work:

```javascript
var macros = {};
katex.renderToString(String.raw`\gdef\foo{1}`, { macros });
katex.renderToString(String.raw`\foo`, { macros })
```

**Comment by edemaine** [comment]
Right, that's the intended behavior -- sorry it took you hours to figure out! Do you have a suggestion on where we (or you via PR) could mention this in the documentation that would have helped you find it sooner?
